src/core/lib/surface/init.c

```diff
@@ -175,14 +176,16 @@ void grpc_shutdown(void) {
       GRPC_EXEC_CTX_INITIALIZER(0, grpc_never_ready_to_finish, NULL);
   gpr_mu_lock(&g_init_mu);
   if (--g_initializations == 0) {
-    grpc_iomgr_shutdown(&exec_ctx);
-    gpr_timers_global_destroy();
-    grpc_tracer_shutdown();
+    grpc_executor_shutdown(&exec_ctx);
```

Comment:
Do we actually want to shut down the executor and timer threads before we shut down plugins?  If the resolver callback is using the executor thread, it probably needs to continue to exist until after the plugins are shut down.

Reply:
This is borrowed from @yang-g 's #11433.